frontend/src/metabase/actions/containers/ActionParametersInputForm/ActionParametersInputForm.tsx

```diff
@@ -1,62 +1,50 @@
-import { useCallback, useMemo, useState, useEffect } from "react";
+import { useCallback, useEffect, useMemo, useState } from "react";
+import { useLatest } from "react-use";
 import { t } from "ttag";
 import _ from "underscore";
 
 import EmptyState from "metabase/components/EmptyState";
 
-import { ActionsApi, PublicApi } from "metabase/services";
-
 import ActionForm from "metabase/actions/components/ActionForm";
-import { getDashboardType } from "metabase/dashboard/utils";
 
 import type {
-  WritebackParameter,
   OnSubmitActionForm,
-  Dashboard,
-  ActionDashboardCard,
   ParametersForActionExecution,
   WritebackAction,
+  WritebackParameter,
 } from "metabase-types/api";
 
 export interface ActionParametersInputFormProps {
   action: WritebackAction;
-  dashboard?: Dashboard;
-  dashcard?: ActionDashboardCard;
   mappedParameters?: WritebackParameter[];
-  dashcardParamValues?: ParametersForActionExecution;
+  initialValues?: ParametersForActionExecution;
+  fetchInitialValues?: () => Promise<ParametersForActionExecution>;
+  shouldPrefetch?: boolean;
   onSubmit: OnSubmitActionForm;
   onSubmitSuccess?: () => void;
   onCancel?: () => void;
 }
 
-const shouldPrefetchValues = (action: WritebackAction) =>
-  action.type === "implicit" && action.kind === "row/update";
-
 function ActionParametersInputForm({
   action,
   mappedParameters = [],
-  dashcardParamValues = {},
-  dashboard,
-  dashcard,
```

Comment:
I refactored `ActionParametersInputForm` so that it no longer depends on `dashboard` and `dashcard`. This way it can be reused in different contexts.